### ISSUE TITLE
TypeScript conflict in templates with stylistic semi: false

### ISSUE THREAD

**Issue by m-shum** [nuxt/eslint#545]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v22.12.0
- Nuxt Version:     3.15.4
- CLI Version:      3.21.1
- Nitro Version:    2.10.4
- Package Manager:  npm@10.9.0
- Builder:          -
- User Config:      modules, devtools, css, compatibilityDate, vite, eslint
- Runtime Modules:  @nuxt/eslint@1.0.1, @vueuse/nuxt@12.6.1, @storyblok/nuxt@6.2.2
- Build Modules:    -
------------------------------

### Package

@nuxt/eslint

### Reproduction

https://codesandbox.io/p/devbox/elegant-sammet-kwkx7j?file=%2Feslint.config.mjs%3A8%2C5

This of course won't show the problem because codesandbox has their own linter that ignores whatever linting setup you add. Please see screenshots below.

### Describe the bug

When setting semicolons to false in stylistic rules, highlighting breaks when using typescript. If a semicolon is added, the highlighting is corrected, but the linter flags it as an error, as per the stylistic rules.

### Additional context

<img width="284" alt="Image" src="https://github.com/user-attachments/assets/b639d044-344d-4631-84f3-b11ecff11d2d" />

<img width="328" alt="Image" src="https://github.com/user-attachments/assets/cc64ff40-8606-48eb-8348-96e64ab20840" />

### Logs

```shell-script

```

**Comment by antfu** [comment]
What do you mean? `semi` is `false` by default. The highlighting error sounds like a bug of Vue language tools but not related to this project. When `semi` is set to `false`, you expect no semi whenever possible, isn't it? I think the rules work as expect.
